civihr_employee_portal/src/View/CustomFieldsMappingFixer.php

```diff
@@ -0,0 +1,146 @@
+        if (!is_array($displayOption)) {
+          continue;
+        }
+        self::processDisplayOptions($displayOption);
+      }
+    }
+  }
+
+  /**
+   * Process all display options for an exported view only when it is related
+   * to a field and the field has a number at the end of its name, which is
+   * the pattern for custom fields from CiviCRM
+   *
+   * @var array
+   *     Display option a exported view structure
+   */
+  static private function processDisplayOptions(&$displayOption) {
+    foreach ($displayOption as &$option) {
+      // only continue if this option is about a field
+      if (!isset($option['field'])) {
+        continue;
+      }
+      $option_id_parts = explode('_', $option['field']);
+      $final_id_part = array_pop($option_id_parts);
+      // it must contain underscores and a number at the end to be checked
+      if (count($option_id_parts) === 0 || !intval($final_id_part)) {
+        continue;
+      }
+      self::processOption($option);
+    }
+  }
+
+  /**
+   * To process a display option for a field in view
+   * @var array $option
+   *     structure that contains an option for a field in view
+   */
+  static private function processOption(&$option) {
+    // checking if this table is part of the civicrm views integration
+    $prefix = self::setCiviCRMtableAndGetPrefix($option['table']);
+    if ( $prefix ) {
+      $columns = self::getColumnsFromTable($prefix.$option['table']);
+      // if the table where the current field belongs exists does not have it
+      // then we can search for similar fields in the same table and make replacements
+      if (!isset($columns[$option['field']])) {
+        $field_name_base = self::removeLastNumberOfString($option['field']);
+        $pos = array_search($field_name_base, $columns);
+        if ($pos !== FALSE) {
+          $option['field'] = array_search($columns[$pos], $columns);
+        }
+      }
+    }
+  }
+
+  /**
+   * Removes the number in end of a string
+   * In this context that number is always separated by an underscore
+   *
+   * @param  string $string
+   *     the string which could have underscores or not
+   *
+   * @return string
+   *     the string without the number at the end,
+   *     if no number found at the end returns the same string
+   */
+  private static function removeLastNumberOfString($string) {
+    $pieces = explode('_', $string);
+    $lastIndex = count($pieces) - 1;
+    return $lastIndex > 0 && intval(array_pop($pieces)) ? implode('_', $pieces) : $string;
+  }
+
+  /**
+   * Return an existing CiviCRM table based on the base name of the table,
+   * which is the name of the table without the number at the end
+   *
+   * @param string $exportedTableWithId
+   *     this is the table name exported by views
+   *
+   * @return string
+   *     the name of the existing table if found or an empty string
+   */
+  private static function setCiviCRMtableAndGetPrefix(&$tableName) {
```

Comment:
Almost every time we have a method with an "and" in its name, it means it does doing two things. A method should only do one thing (that is, have only one responsibility). Please check if we have split this. Perhaps one that sets the tables map and other that gets the prefix from this table